rotkehlchen/accounting/history_base_entries.py

```diff
@@ -0,0 +1,269 @@
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    fee_key = str(HistoryEventType.SPEND) + '__' + str(HistoryEventSubType.FEE)
+    result[fee_key] = BaseEventSettins(
+        taxable=True,
+        count_entire_amount_spend=True,
+        count_cost_basis_pnl=True,
+        method='spend',
+    )
+    renew_key = str(HistoryEventType.RENEW) + '__' + str(HistoryEventSubType.NONE)
+    result[renew_key] = BaseEventSettins(
+        taxable=True,
+        count_entire_amount_spend=True,
+        count_cost_basis_pnl=True,
+        method='spend',
+    )
+    swap_key = str(HistoryEventType.TRADE) + '__' + str(HistoryEventSubType.SPEND)
+    result[swap_key] = BaseEventSettins(
+        taxable=True,
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=True,
+        method='spend',
+        accounting_treatment=TxAccountingTreatment.SWAP,
+    )
+    airdrop_key = str(HistoryEventType.RECEIVE) + '__' + str(HistoryEventSubType.AIRDROP)
+    result[airdrop_key] = BaseEventSettins(
+        taxable=LedgerActionType.AIRDROP in pot.settings.taxable_ledger_actions,
+        # count_entire_amount_spend and count_cost_basis_pnl don't matter for acquisitions.
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    reward_key = str(HistoryEventType.RECEIVE) + '__' + str(HistoryEventSubType.REWARD)
+    result[reward_key] = BaseEventSettins(
+        taxable=True,
+        # count_entire_amount_spend and count_cost_basis_pnl don't matter for acquisitions.
+        count_entire_amount_spend=False,
+        count_cost_basis_pnl=False,
+        method='acquisition',
+    )
+    return result
+
+
+T = TypeVar('T', bound=HistoryBaseEntry)
+
+
+def history_base_entries_iterator(
+        events_iterator: Iterator[AccountingEventMixin],
+        associated_event: T,
+) -> Iterator[T]:
+    """
+    Takes an iterator of accounting events and transforms it into a history base entries iterator.
+    Takes associated event as an argument to be able to log it in case of errors.
+    """
+    for event in events_iterator:
+        if isinstance(event, type(associated_event)) is False:
+            log.error(
+                f'At accounting for event {associated_event.notes} with identifier '
+                f'{associated_event.event_identifier} we expected to take an additional '
+                f'event but found a non {type(associated_event)} event',
+            )
+            return
+
+        yield event  # type: ignore[misc]  # event is guaranteed to be of type T
+
+
+class HistoryBaseEntriesAccountant:
+
+    def __init__(
+            self,
+            evm_accounting_aggregators: 'EVMAccountingAggregators',
+            pot: 'AccountingPot',
+    ) -> None:
+        self.evm_accounting_aggregators = evm_accounting_aggregators
+        self.pot = pot
+        self.tx_event_settings: dict[str, BaseEventSettins] = {}
```

Comment:
This is not just for transactions anymore

```suggestion
        self.event_settings: dict[str, BaseEventSettins] = {}
```